src/sparseml/transformers/text_classification.py

```diff
@@ -563,14 +585,31 @@ def main():
         model.config.label2id = {l: i for i, l in enumerate(label_list)}
         model.config.id2label = {id: label for label, id in config.label2id.items()}
 
-    if data_args.max_seq_length > tokenizer.model_max_length:
+    max_seq_length = data_args.max_seq_length
+    if max_seq_length > tokenizer.model_max_length:
         _LOGGER.warning(
-            f"The max_seq_length passed ({data_args.max_seq_length}) is larger than "
+            f"The max_seq_length passed ({max_seq_length}) is larger than "
             f"the maximum length for the model ({tokenizer.model_max_length}). "
             f"Using max_seq_length={tokenizer.model_max_length}."
         )
+    if teacher_tokenizer and max_seq_length > teacher_tokenizer.model_max_length:
+        _LOGGER.warning(
+            f"The max_seq_length passed ({max_seq_length}) is larger than "
+            "the maximum length for the teacher model "
+            f"({teacher_tokenizer.model_max_length}). "
+            f"Using max_seq_length={teacher_tokenizer.model_max_length}."
+        )
     max_seq_length = min(data_args.max_seq_length, tokenizer.model_max_length)
```

Comment:
Contrary to the warning above, max_seq_length is not taking into consideration teacher_tokenizer.model_max_length

Reply:
good catch, removed